libdnf/sack/query.cpp

```diff
@@ -769,6 +879,64 @@ Query::addFilter(HyNevra nevra, bool icase)
     return 0;
 }
 
+void
+Query::Impl::filterNevraStrict(int cmpType, const char **matches)
+{
+    Pool *pool = dnf_sack_get_pool(sack);
+    std::vector<NevraID> compareSet;
+    const unsigned nmatches = g_strv_length((gchar**)matches);
+    compareSet.reserve(nmatches);
+    NevraID nevraId;
+    for (unsigned int i = 0; i < nmatches; ++i) {
+        const char * nevraPattern = matches[i];
+        if (!nevraPattern)
+            throw std::runtime_error("Query can not accept NULL for STR match");
+        if (nevraId.parse(pool, nevraPattern)) {
+            compareSet.push_back(nevraId);
+        }
+    }
+    Map nevraResult;
+    map_init(&nevraResult, pool->nsolvables);
+
+    if (compareSet.size() > 1) {
+        std::sort(compareSet.begin(), compareSet.end(), nevraIDSorter);
+
+        Id id = -1;
+        while (true) {
+            id = result->next(id);
+            if (id == -1)
+                    break;
+            Solvable* s = pool_id2solvable(pool, id);
+            auto low = std::lower_bound(compareSet.begin(), compareSet.end(), *s,
+                                        nevraCompareLowerSolvable);
+            bool sameNEVRA = low->name == s->name && low->evr == s->evr && low->arch == s->arch;
```

Comment:
You are dereferencing `low` before test `low != compareSet.end()`!
Example of soluton:
```
            if (low != compareSet.end()) {
                bool sameNEVRA = low->name == s->name && low->evr == s->evr && low->arch == s->arch;
                if (sameNEVRA)
                    MAPSET(&nevraResult, id);
            }

```

Reply:
Good point. The part was moved upward due to stile, but not careful move.